### PR TITLE
Improvements in Decryptors:

### PR DESCRIPTION
**Pull request by jwegne**
- Handle Base64 decode errors
- Return the complete string including markers if it could not decoded
- Corrected PKCS#12 to PKCS#8

Signed-off-by: Josef Wegner <josef.wegner@here.com>

### CODE REVIEW

File: xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/PrivateKeyEventDecryptor.java [hunk 1]
```diff
@@ -94,19 +94,30 @@
...
+    if (encoded == null || encoded.equals("")) {
+      return encoded;
+    }
+
+    // we cannot decode the secret if no private key is set.
     if (privateKey == null) {
       return encoded;
     }
+
+    String tmp = encoded.substring(2, encoded.length() - 2);
```
Comment (qGYdXbY2) [inline xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/PrivateKeyEventDecryptor.java:106]: encode.length() > 1  is asserted here

File: xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/KmsEventDecryptor.java [hunk 2]
```diff
@@ -79,14 +79,18 @@ public String decryptAsymmetric(String encoded) {
       return encoded;
     }
 
-    ByteBuffer cipherText = ByteBuffer.wrap(Base64.getDecoder().decode(encoded.getBytes()));
-    DecryptRequest req = new DecryptRequest()
-        .withKeyId(kmsKeyArn)
-        .withEncryptionAlgorithm(ASYMMETRIC_ALGORITHM)
-        .withCiphertextBlob(cipherText);
+    String tmp = encoded.substring(2, encoded.length() - 2);
```
Comment (qGYdXbY2) [inline xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/KmsEventDecryptor.java:82]: encode.length() > 1 is asserted here

Comment (jwegne) [reply xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/KmsEventDecryptor.java:82]: We check now if the string is encrypted. This means it is not null, contains markers at beginning and end, so it is at least 4 chars long.

File: xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/PrivateKeyEventDecryptor.java (hunk 1 above)
Comment (jwegne) [reply xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/PrivateKeyEventDecryptor.java:106]: We check now if the string is encrypted. This means it is not null, contains markers at beginning and end, so it is at least 4 chars long.

**Pull request merged**